workspace/vm/linux/ssh.nix

```diff
@@ -0,0 +1,33 @@
+{ pkgs }:
+
+let
+sshdPatch = pkgs.writeText "sshd_patch.diff" ''
+    *** a/etc/ssh/sshd_config
+    --- b/etc/ssh/sshd_config
+    ***************
+    *** 32 ****
+    ! #PermitRootLogin prohibit-password
+    --- 32 ----
+    ! PermitRootLogin yes
+    ***************
+    *** 57,58 ****
+    ! #PasswordAuthentication yes
+    ! #PermitEmptyPasswords no
+    --- 57,58 ----
+    ! PasswordAuthentication yes
+    ! PermitEmptyPasswords yes
+    ***************
+    *** 116 ****
+    --- 117,118 ----
+    + 
+    + UsePrivilegeSeparation no
+  '';
+in
+pkgs.openssh.overrideAttrs (oldAttrs: {
+    # Disable checks which error
+    doCheck = false;
+
+    postInstall = (oldAttrs.postInstall or "") + ''
+      patch -p1 -d $out < ${sshdPatch}
+    '';
```

Comment:
This works, but it might be simpler to add to `patches` array. The `sshd_config` file is at the root of `$src`. 

Even better, do we have to patch the package itself? The NixOS options that configure sshd work by editing `/etc/ssh/sshd_config` ([reference](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/services/networking/ssh/sshd.nix#L563)). We already pass through the root filesystem, so we could make these changes either in the challenge Dockerfile or in the init script as well. It's fine if we want to keep everything in nix-land though.

Reply:
Yeah, just depends if we want to switch it out of nix or not

Reply:
We definitely want the sshd config to be in nix-land, since this is effectively nothing more than a dependency of the linux vm (to get a shell).

Reply:
> This works, but it might be simpler to add to `patches` array. The `sshd_config` file is at the root of `$src`.

It seems like after some testing that it would require adding more context to patch it which we probably do not want to do